packages/rollup-plugin-import-meta-assets/src/rollup-plugin-import-meta-assets.js

```diff
@@ -0,0 +1,88 @@
+'use strict';
+
+const fs = require('fs');
+const path = require('path');
+const { createFilter } = require('@rollup/pluginutils');
+const { asyncWalk } = require('estree-walker');
+const MagicString = require('magic-string');
+
+function getRelativeAssetPath(node) {
+  const browserPath = node.arguments[0].value;
+  return browserPath.split('/').join(path.sep);
+}
+
+// Looking for this:
+// new URL('./path/to/asset.ext', import.meta.url)
+function isNewUrlImportMetaUrl(node) {
+  return (
+    node.type === 'NewExpression' &&
+    node.callee.type === 'Identifier' &&
+    node.callee.name === 'URL' &&
+    node.arguments.length === 2 &&
+    node.arguments[0].type === 'Literal' &&
+    typeof getRelativeAssetPath(node) === 'string' &&
+    node.arguments[1].type === 'MemberExpression' &&
+    node.arguments[1].object.type === 'MetaProperty' &&
+    node.arguments[1].property.type === 'Identifier' &&
+    node.arguments[1].property.name === 'url'
+  );
+}
+
+function importMetaAssets({ include, exclude, warnOnError, transform } = {}) {
```

Comment:
@LarsDenBakker there are no types right? why does the linting of types does not fail?

Reply:
I have no idea. I'm new here :laughing:
Is there another plugin source I can get inspiration from?

Reply:
potentially here https://github.com/modernweb-dev/web/blob/master/packages/rollup-plugin-copy/src/copy.js